From aa32aef19d3a54a25dc16b3b77e29e902ee8abd7 Mon Sep 17 00:00:00 2001
From: Yi Fan Yu <yifan.yu@windriver.com>
Date: Thu, 1 Apr 2021 13:08:37 -0700
Subject: [PATCH] Skip failing tests due to load variability on YP AB

Skip these tests until AB-INT is solved.

[YOCTO #14296]

Upstream-Status: Inappropriate [OE-Specific]

Signed-off-by: Yi Fan Yu <yifan.yu@windriver.com>

Skip two additional tests due to suspected load variability failures.

[YOCTO #15131]
[YOCTO #15177]

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

Refresh to apply on top of Python 3.14.0.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>
---
 Lib/test/_test_multiprocessing.py | 3 +++
 Lib/test/test_time.py             | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/Lib/test/_test_multiprocessing.py b/Lib/test/_test_multiprocessing.py
index 9a72362c022..008e6a505a5 100644
--- a/Lib/test/_test_multiprocessing.py
+++ b/Lib/test/_test_multiprocessing.py
@@ -722,6 +722,7 @@ def test_close(self):
         close_queue(q)
 
     @support.requires_resource('walltime')
+    @unittest.skip('timing related test, dependent on load')
     def test_many_processes(self):
         if self.TYPE == 'threads':
             self.skipTest('test not appropriate for {}'.format(self.TYPE))
@@ -2345,6 +2346,7 @@ def _test_timeout_f(cls, barrier, results):
         except threading.BrokenBarrierError:
             results.append(True)
 
+    @unittest.skip('timing related test, dependent on load')
     def test_timeout(self):
         """
         Test wait(timeout)
@@ -5459,6 +5461,7 @@ def signal_and_sleep(cls, sem, period):
         time.sleep(period)
 
     @support.requires_resource('walltime')
+    @unittest.skip('timing related test, dependent on load')
     def test_wait_integer(self):
         from multiprocessing.connection import wait
 
diff --git a/Lib/test/test_time.py b/Lib/test/test_time.py
index 5312faa5077..36873d9a400 100644
--- a/Lib/test/test_time.py
+++ b/Lib/test/test_time.py
@@ -548,6 +548,7 @@ def test_monotonic(self):
     def test_perf_counter(self):
         time.perf_counter()
 
+    @unittest.skip('timing related test, dependent on load')
     @unittest.skipIf(
         support.is_wasi, "process_time not available on WASI"
     )
@@ -567,6 +568,7 @@ def test_process_time(self):
         self.assertTrue(info.monotonic)
         self.assertFalse(info.adjustable)
 
+    @unittest.skip('timing related test, dependent on load')
     def test_thread_time(self):
         if not hasattr(time, 'thread_time'):
             if sys.platform.startswith(('linux', 'android', 'win')):
-- 
2.52.0

